compiler/rustc_mir/src/interpret/intrinsics/gc_layout.rs

```diff
@@ -0,0 +1,72 @@
+use rustc_middle::mir::interpret::Allocation;
+use rustc_middle::ty::layout::LayoutCx;
+use rustc_middle::ty::{ParamEnv, Ty, TyCtxt};
+use rustc_span::symbol::sym;
+use rustc_target::abi::{Align, Size};
+use std::mem::size_of;
+
+fn gc_layout(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, param_env: ParamEnv<'tcx>) -> Vec<u8> {
+    let layout = tcx.layout_of(param_env.and(ty)).unwrap();
+    let w_size = Size::from_bytes(size_of::<usize>());
+    let w_align = Align::from_bytes(w_size.bytes()).unwrap();
+
+    if layout.align.abi.bytes() < (w_size.bytes()) {
+        // If an ADT is aligned to a size smaller than 1 word, then it
+        // contains no fields large enough to represent a GC pointer.
+        //
+        // By default, the collector will conservatively scan a struct's
+        // memory, treating it as word-sized chunks. However, since rust is
+        // free to shuffle field order around, any in-memory words which
+        // represent actual pointers would be purely accidental. Such
+        // structs should be ignored completely by the collector.
+        return vec![0; 16]; // empty (bitmap, size) (u64,u64) as byte stream
+    }
+
+    let src_field_iter = ty.ty_adt_def().unwrap().all_fields();
+    let mut bitmap: u64 = 0xFFFFFFFF;
```

Comment:
Do you need twice as many `F`s here? That's only a u32 constant. Could use `u64::MAX`?

Reply:
You're right, `u64::MAX` would be better.

Reply:
Fixed: dfa5cafce3c376f804980b41253f030f2d928bc6